libraries/_helper.rb

```diff
@@ -326,12 +386,31 @@ def upload_cookbook(cookbook)
       end
     end
 
+    # Render a knife config file that points at the new delivery cluster
+    def render_knife_config
+      template File.join(cluster_data_dir, 'knife.rb') do
+        variables lazy {
+          {
+            chef_server_url:      chef_server_url,
+            client_key:           "#{cluster_data_dir}/delivery.pem",
+            analytics_server_url: if analytics_enabled?
```

Comment:
For the optional keys (`analytics_server_url` and `supermarket_site`) can we just not add them to the `knife.rb` at all?